lib/node_modules/@stdlib/console/log-each-map/lib/main.js

```diff
@@ -0,0 +1,160 @@
+var isFunction = require( '@stdlib/assert/is-function' );
+var isCollection = require( '@stdlib/assert/is-collection' );
+var resolveGetter = require( '@stdlib/array/base/resolve-getter' );
+var format = require( '@stdlib/string/format' );
+var logger = require( '@stdlib/console/log' );
+
+
+// MAIN //
+
+/**
+* Inserts array element values into a format string, applies a callback, and prints the result.
+*
+* @param {string} str - format string
+* @param {...(Collection|*)} args - collections or values
+* @param {Function} clbk - callback function
+* @param {*} [thisArg] - callback execution context
+* @throws {TypeError} function must be provided at least 3 arguments
+* @throws {TypeError} first argument must be a string
+* @throws {TypeError} callback must be a function
+* @throws {RangeError} provided collections must have the same length
+* @returns {void}
+*
+* @example
+* function add( x, y ) {
+*     return x + y;
+* }
+*
+* var x = [ 1, 2, 3 ];
+* var y = [ 4, 5, 6 ];
+*
+* logEachMap( '%d + %d = %d', x, y, add );
+* // e.g., => '1 + 4 = 5\n2 + 5 = 7\n3 + 6 = 9\n'
+*/
+function logEachMap() {
+	var strides;
+	var offsets;
+	var getters;
+	var thisArg;
+	var cbArgs;
+	var values;
+	var nargs;
+	var args;
+	var clbk;
+	var str;
+	var len;
+	var v;
+	var s;
+	var n;
+	var i;
+	var j;
+
+	nargs = arguments.length;
+	if ( nargs < 3 ) {
+		throw new TypeError( 'invalid invocation. Must provide a format string, at least one array-like object, and a callback function.' );
+	}
+
+	str = arguments[ 0 ];
+	if ( !isString( str ) ) {
+		throw new TypeError( format( 'invalid argument. First argument must be a string. Value: `%s`.', str ) );
+	}
+
+	if ( isFunction( arguments[ nargs-1 ] ) ) {
+		clbk = arguments[ nargs-1 ];
+		thisArg = null;
+		n = nargs - 2;
+	} else {
+		clbk = arguments[ nargs-2 ];
+		if ( !isFunction( clbk ) ) {
+			throw new TypeError( format( 'invalid argument. Callback argument must be a function. Value: `%s`.', clbk ) );
+		}
+		thisArg = arguments[ nargs-1 ];
+		n = nargs - 3;
+	}
+
+	getters = [];
+	strides = [];
+	args = [];
+	for ( i = 1; i < n+1; i++ ) {
+		v = arguments[ i ];
+		if ( isCollection( v ) ) {
+			getters.push( resolveGetter( v ) );
+			args.push( v );
+			strides.push( 1 );
+			len = v.length;
+			i += 1;
+			break;
+		} else {
+			v = [ v ];
+			getters.push( resolveGetter( v ) );
+			args.push( v );
+			strides.push( 0 );
+		}
+	}
+	if ( len === void 0 ) {
+		len = 1;
+	}
+	for ( ; i < n+1; i++ ) {
+		v = arguments[ i ];
+		if ( isCollection( v ) ) {
+			if ( v.length !== len ) {
+				throw new RangeError( 'invalid argument. Provided collections must have the same length.' );
+			}
+			s = 1;
+		} else {
+			v = [ v ];
+			s = 0;
+		}
+		getters.push( resolveGetter( v ) );
+		args.push( v );
+		strides.push( s );
+	}
+
+	values = [ str ];
+	offsets = [];
+	for ( i = 1; i < n+2; i++ ) {
+		values.push( null );
+		offsets.push( 0 );
+	}
+
+	cbArgs = [];
+	cbArgs[ n + 1 ] = args;
```

Comment:
Don't do this. This automatically puts arrays into dictionary mode. You need to explicitly allocate to ensure a contiguous generic array. Feel free to use https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/array/base/nulls.